processor/metricstransformprocessor/datapoint_aggregation.go

```diff
@@ -0,0 +1,259 @@
+			for _, p := range distPoints {
+				newPoints = append(newPoints, &metricspb.Point{
+					Timestamp: timestamp,
+					Value:     p,
+				})
+			}
+
+		}
+	}
+	return newPoints
+}
+
+// computeInt64 attempts to merge int64 points into one point based on the provided aggrType
+// Returns the aggregated int64 value
+func (mtp *metricsTransformProcessor) computeInt64(points []*metricspb.Point, aggrType AggregationType) *metricspb.Point_Int64Value {
+	intVal := int64(0)
+	if len(points) > 0 {
+		intVal = points[0].GetInt64Value()
+	}
+	for _, p := range points[1:] {
+		if aggrType == Sum || aggrType == Average {
+			intVal += p.GetInt64Value()
+		} else if aggrType == Max {
+			intVal = int64(math.Max(float64(intVal), float64(p.GetInt64Value())))
+		} else if aggrType == Min {
+			intVal = int64(math.Min(float64(intVal), float64(p.GetInt64Value())))
+		}
+	}
+	if aggrType == Average {
+		intVal /= int64(len(points))
+	}
+	return &metricspb.Point_Int64Value{Int64Value: intVal}
+}
+
+// computeDouble attempts to merge double points into one point based on the provided aggrType
+// Returns the aggregated double value
+func (mtp *metricsTransformProcessor) computeDouble(points []*metricspb.Point, aggrType AggregationType) *metricspb.Point_DoubleValue {
+	doubleVal := float64(0)
+	if len(points) > 0 {
+		doubleVal = points[0].GetDoubleValue()
+	}
+	for _, p := range points[1:] {
+		if aggrType == Sum || aggrType == Average {
+			doubleVal += p.GetDoubleValue()
+		} else if aggrType == Max {
+			doubleVal = math.Max(doubleVal, p.GetDoubleValue())
+		} else if aggrType == Min {
+			doubleVal = math.Min(doubleVal, p.GetDoubleValue())
+		}
+	}
+	if aggrType == Average {
+		doubleVal /= float64(len(points))
+	}
+	return &metricspb.Point_DoubleValue{DoubleValue: doubleVal}
+}
+
+// computeDistribution attempts to merge distribution points into one point
+// Returns the aggregated distribution value or values if there are mismatching bounds
+func (mtp *metricsTransformProcessor) computeDistribution(points []*metricspb.Point) []*metricspb.Point_DistributionValue {
+	distVals := make([]*metricspb.Point_DistributionValue, 0)
+	boundsToPoints := mtp.groupPointsByBounds(points)
+	for _, v := range boundsToPoints {
+		var distVal *metricspb.DistributionValue
+		for _, p := range v {
+			if distVal == nil {
+				distVal = p.GetDistributionValue()
+				continue
+			}
+			distVal = mtp.computeDistVals(distVal, p.GetDistributionValue())
+		}
+		distVals = append(distVals, &metricspb.Point_DistributionValue{DistributionValue: distVal})
+	}
+	return distVals
+}
+
+// groupPointsByBounds groups distribution value points by the bounds to further determine the aggregatability because distribution value points can only be aggregated if they have match bounds
+// Returns a map with groups of aggregatable distribution value points
+func (mtp *metricsTransformProcessor) groupPointsByBounds(points []*metricspb.Point) map[string][]*metricspb.Point {
```

Comment:
Should you check for the case where the bounds are different and log a warning?

Reply:
So in this function, I am grouping points by bounds, so the points with different bounds will not be grouped together as an aggregatable set. Also, I have added a log when there are mismatching bounds.